crates/integration_tests/testdata/pyiceberg/provision.py

```diff
@@ -0,0 +1,80 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+from datafusion import SessionContext
+from pyiceberg.catalog import load_catalog
+import pyarrow.parquet as pq
+
+# Generate a table with various types in memory and dump to a Parquet file
+ctx = SessionContext()
+ctx.sql("""
+CREATE TABLE types_test (
```

Comment:
lets generate these values in pyiceberg/pyarrow. it'll help us figure out the supportability of each types which is where the E2E test is useful.

Here are the supported types we use in pyiceberg's integration test 
https://github.com/apache/iceberg-python/blob/acd6f5a8a19db709e835e2686b87d4db3dca254f/tests/conftest.py#L304-L349

Reply:
Sounds good, I just switched to pyarrow to generate the Parquet file from which the table is constructed.

To be clear, if #850 gets accepted and merged, and if folks don't feel comfortable including pyiceberg in the testing scope for now, then this PR can just remain as an example for the issue in #813 (it still exercises it as demonstrated at the very end of the integration test).